net/instaweb/rewriter/rewrite_options.cc

```diff
@@ -4432,6 +4468,108 @@ bool RewriteOptions::MergeOK() const {
 }
 #endif
 
+bool RewriteOptions::ValidateConfiguredHttpHeader(const GoogleString& name,
+                                                  const GoogleString& value,
+                                                  GoogleString* error_message) {
+  DCHECK(error_message != NULL);
+
+  if (error_message == NULL) {
+    return false;
+  }
+
+  // From <https://www.ietf.org/rfc/rfc2616.txt>
+  // token          = 1*<any CHAR except CTLs or separators>
+  //       separators     = "(" | ")" | "<" | ">" | "@"
+  //                      | "," | ";" | ":" | "\" | <">
+  //                      | "/" | "[" | "]" | "?" | "="
+  //                      | "{" | "}" | SP | HT
+  const GoogleString separators("()<>@,;:\\\"/[]?={} \t");
+
+  // Reject empty field names. Empty field-values are acceptable.
+  // Impose a max length of 1024 on both field-name and field-value.
+  if (name.size() == 0) {
+    *error_message = "Empty field name not allowed";
+    return false;
+  }
+  if (name.size() > 1024) {
+    *error_message = "Field name too long";
+  }
+  if (value.size() > 1024) {
+    *error_message = "Field value too long";
+    return false;
+  }
+
+  for (size_t i = 0; i < name.size(); ++i) {
+    if (!IsNonControlAscii(name[i])) {
+      *error_message = "Invalid character in field name";
+      return false;
+    }
+    if (separators.find(name[i]) != GoogleString::npos) {
```

Comment:
This looks slow. How about static-initing an ascii-ordered char[] kSeparators and using binary-search on that, or use a 256-element bool array that's initialized where you set up the string-vector array above.